hadoop-tools/hadoop-fs2img/src/main/java/org/apache/hadoop/hdfs/server/namenode/FSTreeWalk.java

```diff
@@ -71,14 +95,24 @@ private FSTreeIterator() {
     }
 
     FSTreeIterator(TreePath p) {
+      AclStatus acls = null;
+      Path remotePath = p.getFileStatus().getPath();
+      try {
+        acls = getAclStatus(fs, remotePath);
+      } catch (IOException e) {
+        LOG.warn(
+            "Got exception when trying to get remote acls for path {} : {}",
+            remotePath, e.getMessage());
+      }
       getPendingQueue().addFirst(
-          new TreePath(p.getFileStatus(), p.getParentId(), this, fs));
+          new TreePath(p.getFileStatus(), p.getParentId(), this, fs, acls));
```

Comment:
Extract.

Reply:
Do we support adding null to the acls?
Is there a finer grain exception then IOException?

Reply:
ACL can be `null`.
`IOException` comes from the `FileSystems` API, `AclStatus getAclStatus(Path path) throws IOException`